tests/unittests/GraphOptzTest.cpp

```diff
@@ -28,11 +28,34 @@ using namespace glow;
 
 class GraphOptz : public ::testing::Test {
 public:
-  GraphOptz() { F_ = mod_.createFunction("main"); }
+  GraphOptz() {
+    mod_ = &EE_.getModule();
+    originalF_ = mod_->createFunction("main");
+  }
 
 protected:
-  Module mod_;
-  Function *F_;
+  void checkNumericalEquivalence() {
+    // Check that the function and its optimized complement exist.
+    EXPECT_TRUE(originalF_);
+    EXPECT_TRUE(optimizedF_);
+
+    // Clone bindings to use for original and optimized functions.
+    PlaceholderBindings originalBindings = bindings_.clone();
+    PlaceholderBindings optimizedBindings = bindings_.clone();
+
+    // Compile and run functions.
+    EE_.compile(CompilationMode::Infer);
+    EE_.run(originalBindings, originalF_->getName());
+    EE_.run(optimizedBindings, optimizedF_->getName());
+
+    // Compare outputs.
+    EXPECT_TRUE(
+        PlaceholderBindings::compare(&originalBindings, &optimizedBindings));
+  }
+
+  ExecutionEngine EE_;
+  Module *mod_{nullptr};
```

Comment:
Why do you need `mod_` to be a pointer rather than a reference?

Reply:
Now that there is an `ExecutionEngine` in this class, If we make `mod_`  a reference, it must be initialized in the initializer list of the constructor of this class, i.e.:
```
GraphOptz() : mod_(EE_.getModule()) { ... }
```
Is this defined behaviour? Will `EE_` be initialized before `mod_` even though it doesn't appear in the initializer list?

Reply:
According to the standard:
```
The order of member initializers in the list is irrelevant: the actual order of initialization is as follows:

1) If the constructor is for the most-derived class, virtual base classes are initialized in the order in which they appear in depth-first left-to-right traversal of the base class declarations (left-to-right refers to the appearance in base-specifier lists)
2) Then, direct base classes are initialized in left-to-right order as they appear in this class's base-specifier list
3) Then, non-static data members are initialized in order of declaration in the class definition.
4) Finally, the body of the constructor is executed
```

So, I think you should be fine as long as `EE_` is declared before `mod_`.